packages/core/src/app-differ.ts

```diff
@@ -29,7 +29,7 @@ export default class AppDiffer {
             return {
               mayChange: addon.mayRebuild,
               walk() {
-                return walkSync.entries(definitelyDir);
+                return walkSync.entries(definitelyDir, { ignore: ['**/.*'] });
```

Comment:
Hmm, seems _vaguely_ odd to somewhat arbitrarily exclude all dot files. Does this affect build output (e.g. will modules be _missing_ that would have been present before)?

Reply:
I think we *do* need some special handling for dotfiles. Stock ember-cli refuses to include any dotfiles in the app, even if they have `.js` extensions and are sitting in directories that are normally part of the app.

But I agree this may be the wrong place to do it. It's not clear that we don't want the files present, it's more that we don't want them to be treated as implicit-modules.

Reply:
@rwjblue I tested to confirm it's true, but maybe you know where it actually happens: why does `app/whatever.js` appear in `requirejs.entries` but `app/.whatever.js` does not?

Reply:
let me do a quick debugging.

Reply:
So @thoov and myself did some debugging, it appears that the first thing to filter out dotfiles currently is broccoli-concat.

specifically here:
https://github.com/broccolijs/broccoli-concat/blob/d08f478100ee920e7cac7a32accdd158f35fde08/concat.js#L228

As ` this.allInputFiles` contains `['app-name/**/*.js']` which does not match dotfiles.

```
> walkSync('.', ['dot-dot-dot/**/*.js']).filter(x => x.includes('bar') || x.includes('foo'))
[]
> walkSync('.', ['dot-dot-dot/**/*']).filter(x => x.includes('bar') || x.includes('foo'))
[]
> walkSync('.', ['dot-dot-dot/']).filter(x => x.includes('bar') || x.includes('foo'))
[]
> walkSync('.', ['dot-dot-dot/.*']).filter(x => x.includes('bar') || x.includes('foo'))
[ 'dot-dot-dot/.bar', 'dot-dot-dot/.foo.js' ]
```

Reply:
parity with classic builds may be tricky, since we must let the "babel" step see them, but not let the bundling step consider them as entry point. That being said, maybe there is an better place to put this logic...

Reply:
Chated with @ef4, TL;DR:

```
* all dotfiles are not considered as implicit-modules
* this PR is slightly overzealous.
```

Reply:
actually: https://github.com/embroider-build/embroider/blob/db57ada3aed6bd2fca6369f94856b89fc32250db/packages/compat/src/v1-addon.ts#L452 may be more relevant